packages/devtools_app/lib/src/flutter/banner_messages.dart

```diff
@@ -0,0 +1,304 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+
+import '../auto_dispose.dart';
+import 'auto_dispose_mixin.dart';
+import 'common_widgets.dart';
+import 'controllers.dart';
+import 'screen.dart';
+import 'theme.dart';
+import 'utils.dart';
+
+const _runInProfileModeDocsUrl =
+    'https://flutter.dev/docs/testing/ui-performance#run-in-profile-mode';
+
+const _profileGranularityDocsUrl =
+    'https://flutter.dev/docs/development/tools/devtools/performance#profile-granularity';
+
+class BannerMessagesController implements DisposableController {
+  final _dismissedMessageIds = <String>{};
+
+  final _refreshMessagesController = StreamController<Null>.broadcast();
```

Comment:
It would be nice to do something more idiomatic than a stream with only null elements. 
For example, a ValueListenable<List<IdMessage>> or a ChangeNotifier could be cleaner.
https://api.flutter.dev/flutter/foundation/ChangeNotifier-class.html

Reply:
done - using inherited widget